x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -265,7 +265,9 @@ export function getIndexPatternDatasource({
     getMetaData(state: IndexPatternPrivateState) {
       return {
         filterableIndexPatterns: _.uniq(
-          Object.values(state.layers).map(layer => layer.indexPatternId)
+          Object.values(state.layers)
```

Comment:
If we change the `filterableIndexPatterns` key to hold the name instead of the id, we also have to change that in `embeddable.tsx`

Reply:
This turned into a larger change because the IDs and titles are used for different purposes. IDs are used by the embeddable to load an index pattern service by ID. Titles are used by the QueryBar and QueryBarInput components which have a different loading mechanism.